File: villager/src/main/java/io/charlescd/villager/infrastructure/persistence/ModuleRepository.java
/*
 * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.charlescd.villager.infrastructure.persistence;

import io.quarkus.hibernate.orm.panache.PanacheRepository;

import javax.enterprise.context.ApplicationScoped;
import java.util.List;
import java.util.Optional;

@ApplicationScoped
public class ModuleRepository implements PanacheRepository<ModuleEntity> {

    public List<ModuleEntity> findByJobId(String ciJobId) {
        return find("ciJobId", ciJobId).list();
    }

    public Optional<ModuleEntity> findByNameAndTagNameAndStatus(String name, String tagName, ModuleBuildStatus status) {
        return Optional.ofNullable(find("name = ?1 and tagName = ?2 and status = ?3", name, tagName, status).firstResult());
    }

    public List<ModuleEntity> findModulesByStatus(ModuleBuildStatus status) {
        return find("status = ?1", status).list();
    }

    public List<ModuleEntity> findByBuildId(String buildId) {
        return find("build_id = ?1", buildId).list();
    }

    public void persist(List<ModuleEntity> moduleEntityList) {
        persist(moduleEntityList.stream());
    }

}

File: moove/domain/src/main/kotlin/io/charlescd/moove/domain/repository/ModuleRepository.kt
/*
 *
 *  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
 *  *
 *  * Licensed under the Apache License, Version 2.0 (the "License");
 *  * you may not use this file except in compliance with the License.
 *  * You may obtain a copy of the License at
 *  *
 *  *     http://www.apache.org/licenses/LICENSE-2.0
 *  *
 *  * Unless required by applicable law or agreed to in writing, software
 *  * distributed under the License is distributed on an "AS IS" BASIS,
 *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  * See the License for the specific language governing permissions and
 *  * limitations under the License.
 *
 */

package io.charlescd.moove.domain.repository

import io.charlescd.moove.domain.Component
import io.charlescd.moove.domain.Module
import io.charlescd.moove.domain.Page
import io.charlescd.moove.domain.PageRequest
import java.util.*

interface ModuleRepository {

    fun save(module: Module): Module

    fun update(module: Module): Module

    fun delete(id: String, workspaceId: String)

    fun addComponents(module: Module)

    fun removeComponents(module: Module)

    fun updateComponent(component: Component)

    fun find(id: String): Optional<Module>

    fun find(id: String, workspaceId: String): Optional<Module>

    fun findByWorkspaceId(workspaceId: String, name: String?, pageRequest: PageRequest): Page<Module>

    fun findByIds(ids: List<String>): List<Module>

     fun findByComponentNameAndWorkspaceId(workspaceId: String, name: String): Module?
}
